compilers/concrete-optimizer/concrete-optimizer/src/optimization/dag/multi_parameters/analyze.rs

```diff
@@ -181,6 +187,118 @@ impl VariancedDag {
         }
     }
 
+    fn apply_external_partition_input_variance(&mut self) {
+        for (i, op) in self.dag.operators.clone().iter().enumerate() {
+            if let Operator::Input { .. } = op {
+                let partition_index = self.partitions.instrs_partition[i].instruction_partition;
+                if self
+                    .partitions
+                    .p_cut
+                    .is_external_partition(&partition_index)
+                {
+                    let partitions = self.partitions.clone();
+                    let max_variance = partitions.p_cut.external_partitions
+                        [partition_index.0 - partitions.p_cut.n_internal_partitions()]
+                    .max_variance;
+
+                    let mut input = self.get_operator_mut(OperatorIndex(i));
+                    let mut variances = input.variance().clone();
+                    variances.vars[partition_index.0] = SymbolicVariance::input_external_partition(
+                        partitions.nb_partitions,
+                        partition_index,
+                        max_variance,
```

Comment:
Oups, this is in absolute scale no ? Needs to be rescale using the pbs noise level of that partition.
max_variance / pbs_noise